chain/consensus/actors/atomic-replace/replace.go

```diff
@@ -0,0 +1,264 @@
+		if ok {
+			rt.Abortf(exitcode.ErrIllegalState, "address already owning something")
+		}
+		own.M[rt.Caller().String()], err = abi.CidBuilder.Sum([]byte(params.Seed))
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "error computing cid")
+		st.storeOwners(rt, own)
+	})
+
+	return nil
+}
+
+type ReplaceParams struct {
+	Addr address.Address
+}
+
+func (a ReplaceActor) Replace(rt runtime.Runtime, params *ReplaceParams) *abi.EmptyValue {
+	rt.ValidateImmediateCallerAcceptAny()
+
+	var st ReplaceState
+	rt.StateTransaction(&st, func() {
+		ValidateLockedState(rt, &st)
+		own, err := st.UnwrapOwners()
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "error unwrapping lockable state")
+		_, ok1 := own.M[rt.Caller().String()]
+		_, ok2 := own.M[params.Addr.String()]
+		if !ok1 || !ok2 {
+			rt.Abortf(exitcode.ErrIllegalState, "one (or both) parties don't have an asset to replace")
+		}
+		// Replace
+		own.M[rt.Caller().String()], own.M[params.Addr.String()] =
+			own.M[params.Addr.String()], own.M[rt.Caller().String()]
+		st.storeOwners(rt, own)
+	})
+
+	return nil
+}
+
+///////
+// Atomic function definitions.
+//////
+func (a ReplaceActor) Lock(rt runtime.Runtime, params *atomic.LockParams) *atomic.LockedOutput {
+	// Anyone can lock the state
+	rt.ValidateImmediateCallerAcceptAny()
+
+	var st ReplaceState
+	rt.StateTransaction(&st, func() {
+		switch params.Method {
+		case 5:
+			builtin.RequireNoErr(rt, st.Owners.LockState(), exitcode.ErrIllegalArgument, "error locking state")
+		default:
+			rt.Abortf(exitcode.ErrIllegalArgument, "provided method doesn't support atomic execution. No need to lock")
+		}
+	})
+
+	c, err := st.Owners.Cid()
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "error computing Cid for locked state")
+	return &atomic.LockedOutput{Cid: c}
+}
+
+func (st *ReplaceState) unlock(rt runtime.Runtime) {
+	builtin.RequireNoErr(rt, st.Owners.UnlockState(), exitcode.ErrIllegalArgument, "error unlocking state")
+}
+
+func (a ReplaceActor) Merge(rt runtime.Runtime, params *atomic.MergeParams) *abi.EmptyValue {
+	// Only system actor can trigger this function.
+	rt.ValidateImmediateCallerIs(builtin.SystemActorAddr)
+	var st ReplaceState
+	rt.StateTransaction(&st, func() {
+		merge := &Owners{}
+		err := atomic.UnwrapMergeParams(params, merge)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "error unwrapping output from mergeParams")
+		st.merge(rt, merge)
+	})
+
+	return nil
+}
+
+func (a ReplaceActor) Unlock(rt runtime.Runtime, params *atomic.UnlockParams) *abi.EmptyValue {
+	// Only system actor can trigger this function.
+	rt.ValidateImmediateCallerIs(builtin.SystemActorAddr)
+
+	var st ReplaceState
+	rt.StateTransaction(&st, func() {
+		output := &Owners{}
+		err := atomic.UnwrapUnlockParams(params, output)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "error unwrapping output from unlockParams")
+		switch params.Params.Method {
+		case 5:
```

Comment:
What is `5`? Can we create a constant or something similar?

Reply:
Oh, good catch. I created the constant but forgot to update it there. Thanks!